python/semantic_kernel/kernel.py

```diff
@@ -63,20 +64,21 @@
 
 
 class Kernel:
-    _plugin_collection: PluginCollectionBase
+    # TODO: pydantic-ify these fields
+    _plugins: KernelPluginCollection
```

Comment:
no time like the present, `plugins: KernelPluginCollection = Field(default_factory=KernelPluginCollection)` and then remove the property